Framework/Algorithms/src/IQTransform.cpp

```diff
@@ -215,9 +217,9 @@ void IQTransform::guinierRods(API::MatrixWorkspace_sptr ws) {
  * number
  */
 void IQTransform::guinierSheets(API::MatrixWorkspace_sptr ws) {
-  MantidVec &X = ws->dataX(0);
-  MantidVec &Y = ws->dataY(0);
-  MantidVec &E = ws->dataE(0);
+  auto &X = ws->mutableX(0);
+  auto &Y = ws->mutableY(0);
+  auto &E = ws->mutableE(0);
```

Comment:
@LamarMoore Right now we do not support cross-type operations in that PR, I think. That is, right now you cannot do

```cpp
ws->mutable(0) /= ws->mutableY(0);
```
To me it is not obvious whether this is something we want, or whether the added dangers outweigh the benefits, losing some type safety. What do you think?